app/scripts/controllers/transactions/tx-state-manager.js

```diff
@@ -159,7 +159,12 @@ class TransactionStateManager extends EventEmitter {
         transactions.splice(index, 1)
       }
     }
-    transactions.push(txMeta)
+    const newTxIndex = transactions.findIndex((metaTx) => {
```

Comment:
Note to other reviewers: this change, along with above changes in `transactions/index.js` is only needed to ensure consistency of transaction order. This was an issue for batch transactions and the same change is included in https://github.com/MetaMask/metamask-extension/pull/7473

It was also needed to get my local e2e tests passing. Perhaps we should make a PR with just this change against develop, instead of including it in this PR.

Reply:
Change now made in PR against `develop`: https://github.com/MetaMask/metamask-extension/pull/7484

I will remove it from here once that is merged to develop, and then this branch can be rebased (although, that can also happen after this is merged to `LoginPerSite`)